projects/scripts/adi_project_xilinx.tcl

```diff
@@ -372,10 +372,10 @@ proc adi_project_run {project_name} {
   file mkdir $project_name.sdk
 
   if [expr [string match *VIOLATED* $[report_timing_summary -return_string]] == 1] {
-    file copy -force $project_name.runs/impl_1/system_top.sysdef $project_name.sdk/system_top_bad_timing.hdf
+    write_hw_platform -fixed -force  -include_bit -file $project_name.sdk/system_top_bad_timing.xsa
     return -code error [format "ERROR: Timing Constraints NOT met!"]
   } else {
-    file copy -force $project_name.runs/impl_1/system_top.sysdef $project_name.sdk/system_top.hdf
+    write_hw_platform -fixed -force  -include_bit -file $project_name.sdk/system_top.xsa
```

Comment:
Perhaps we should add a version check here and keep both methods so we can still compile til the end with 2019.1

Keeping the filename 'hdf' would save us some effort from updating the infrastructure scripts. Is there any reason to go with the xsa extension ? 

Thanks.

Reply:
Initially, @sarpadi did that way, but I downvoted it. It freaked me out of the idea to support multiple versions. I'm afraid that some will get greedy, and it will become a habit. 😓 I still believe that we should support one version per release/branch all the time. Maybe I'm just too paranoid or suffered too much because of the tool upgrades.

Reply:
If you see it that way then we should promote the version mismatch check from CRITICAL WARNING to  ERROR and fail early not after tens of minutes of compilation.

Reply:
Really good point! I'm supporting that change.

Reply:
In this way, should I stick to 2019.1 or waiting for migration to 2020.1 or newer vitis tools in using analog HDL?

Reply:
If possible, stick to the latest tools version we support, as different tools may have slightly different behaviors and create issues at system level.